jest.config.ts

```diff
@@ -0,0 +1,10 @@
+import type { Config } from '@jest/types';
+
+const config: Config.InitialOptions = {
+  // Run tests from one or more projects
+  // This is all we need in this file because each package has its own Jest
+  // config
+  projects: ['<rootDir>/packages/*'],
```

Comment:
There isn't a whole lot of documentation about this option other than the [corresponding section in the Jest docs]( https://jestjs.io/docs/configuration#projects-arraystring--projectconfig), but from what I've been able to tell this will run the tests for each directory in `packages` using a fresh runner as if we'd run `jest packages/<package-name>` for each package. If we do not do this and just run all test files across the entire project, Jest will use `tsconfig.json` when compiling each TypeScript file thanks to `ts-jest`. This is a problem because [`ts-jest` doesn't support TypeScript project references](https://github.com/kulshekhar/ts-jest/issues/1648#issuecomment-661874589) and so it isn't able to take each package's `tsconfig.json` into consideration. Specifically, the `paths` option is either ignored or isn't evaluated in the context of each package but in the context of the root project, which results in imports for other packages (e.g. how `ens-controller` imports `@metamask/base-controller`) to not resolve correctly.

By running the tests for each package in their own Jest runner, we can instruct `ts-jest` to use the `tsconfig.json` at the package level and not the root level, which fixes the `paths` resolution.